go/visitorgen/ast_walker.go

```diff
@@ -0,0 +1,130 @@
+/*
```

Comment:
Could do. I was thinking I wanted to use `visitorgen` for our Primitive plans as well. We don't rewrite them today, but we do visit them using the [`Find`](https://github.com/vitessio/vitess/blob/master/go/vt/vtgate/engine/primitive.go#L130) method, and I foresee rewriting of plans in our future as well.

Never mind - it will be easy to move it somewhere else when we do decide to start using it for more than the ast. For now, I'll move it into the `sqlparser` package